src/parser/expressions_parser.rs

```diff
@@ -211,28 +215,28 @@ fn parse_unary_expression(lexer: &mut ParseSession) -> AstStatement {
             },
 
             _ => AstStatement::UnaryExpression {
-                operator,
+                operator: *operator,
                 value: Box::new(expression),
                 location,
                 id: lexer.next_id(),
             },
         }
-    } else {
-        parse_parenthesized_expression(lexer)
-    }
+    })
 }
 
 // PARENTHESIZED (...)
 fn parse_parenthesized_expression(lexer: &mut ParseSession) -> AstStatement {
-    match lexer.token {
+    let result = match lexer.token {
         KeywordParensOpen => {
             lexer.advance();
             super::parse_any_in_region(lexer, vec![KeywordParensClose], |lexer| {
                 parse_expression(lexer)
             })
         }
         _ => parse_leaf_expression(lexer),
-    }
+    };
+    // we might deal with a deref after a paren-expr
+    parse_access_modifiers(lexer, result).unwrap()
```

Comment:
try not to unwrap, if there is a problem report it using the diagnostics reported nested in the lexer and then return an empty statement.